plonk-core/src/constraint_system/composer.rs

```diff
@@ -197,6 +215,7 @@ where
             q_variable_group_add: Vec::with_capacity(expected_size),
             q_lookup: Vec::with_capacity(expected_size),
             public_inputs: PublicInputs::new(),
+            intended_pi_pos: Vec::with_capacity(expected_size),
```

Comment:
I guess in most cases it's quite unlikely that there will be that many public inputs. So the first thought was to choose some smaller value. But looking at all the other things allocating that same capacity, I doubt it will change anything at all, so guess it's not an issue ¯\_(ツ)_/¯

Reply:
You're right! The size of this will be a very small fraction of the expected size in most cases. I'll update with  `new()` instead.